discovery.md

```diff
@@ -0,0 +1,448 @@
+  automatically expire.
+- Constraints:
+  - OPTIONAL
+  - If present, must be a positive Integer >= 1.
+- Examples:
+  - `604800` - producer will expire subscriptions in 7 days
+
+#### authscope
+
+- Type: `String`
+- Description: Authorization scope required for creating subscriptions.
+  The actual meaning of this field is determined on a per-producer basis.
+- Constraints:
+  - OPTIONAL
+- Example:
+  - "storage.read"
+
+### Query API
+
+The query operation MUST be supported by compliant Event Producers. It is used
+by the Event Subscriber to get information about the events that the Event
+Producer produces. To retrieve the list of events produced by an Event Producer,
+the following data is to be provided. The query operation takes a key/value map
+(`config`) of type `string` to `string`.
+
+Each query parameter that is provided is an `and` operation. Each parameter
+MUST support exact match and prefix matching operations. Prefix matching uses
+the syntax of `prefix*` where the `*` character represents the end of the
+prefix.
+
+#### Query API keys
+
+##### producer
+
+- Type: `String`
+- Description: The name of the [producer](#producer) that is offering this
+  events.
+- Constraints:
+  - OPTIONAL
+  - If present, must be non-empty
+- Examples:
+  - "Github"
+  - "Awesome Cloud Storage"
+  - "Awesome Cloud\*"
+
+##### source
+
+- Type: `String`
+- Description: The specific [source](#source) to query for.
+- Constraints:
+  - OPTIONAL
+  - If present, must be non-empty
+- Examples:
+  - "/sensors/\*"
+  - "/sensors/tn-1234567/alerts"
+
+##### type
+
+- Type: `String`
+- Description: The specific [type](#type) to query for.
+- Constraints:
+  - OPTIONAL
+  - If present, must be non-empty
+- Examples:
+  - "com.github.pull.create"
+  - "com.github.pull.\*"
+
+##### expandsources
+
+- Type: boolean
+- Description: If true, the `source` response attribute will be expanded
+  for each producer entity that matches the query.
+- Constraints:
+  - OPTIONAL
+  - If absent, a default value of `false` is used.
+
+## Example Usage
+
+For these examples, we'll assume discovery is being served from an events broker
+that has access to events from multiple producer systems. Specifically, we will
+show these producers, each with the specified event types.
+
+* "Git Source Control"
+ * `git.pullrequest`
+ * `git.commit`
+ * `git.newissue`
+* "Cloud Storage"
+ * `cloud.storage.object.create`
+ * `cloud.storage.object.delete`
+ * `cloud.storage.object.update`
+* "Cloud RDBMS"
+ * `cloud.rdbms.row.insert`
+ * `cloud.rdbms.row.delete`
+ * `cloud.rdbms.table.create`
+ * `cloud.rdbms.table.drop`
+
+### Query for available producers.
+
+In order to configure a subscription, a user wants to browse available events
+on this broker. We start by listing the available producers, an empty discovery
+query and `expandsources` set to `false`
+
+`/discovery?expandsources=false`
+
+The returned data would correspond to this table. Some attributes are omitted.
```

Comment:
Is it fair to assume that the response will have all the attributes of the producer entity?

Reply:
removed sources